server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/TransformModelNodeVisitor.java

```diff
@@ -213,9 +213,11 @@ private void handleOperation(Operation operation, String serverName, String conf
         for (String dependency : operation.getDependencies()) {
             String cfnSource = getFileURL(cfnModule.getBucketName(), dependency);
 
-            logger.debug("Marking " + dependency + " as file to be uploaded.");
+            logger.debug("Marking '{}' as file to be uploaded.", dependency);
             cfnModule.putFileToBeUploaded(dependency);
-
+            logger.debug("Marking '{}' as instance in need of authentication.", serverName);
+            cfnModule.putAuthentication(serverName);
```

Comment:
Maybe check first if the `serverName` is already maked for authentication. But I guess it is a set so it doesn't matter if you add it multiple times

Reply:
You're right, even if it doesn't matter, it might be good to know for debugging reasons.